src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/request/CacheRefreshRequest.java

```diff
@@ -62,8 +64,9 @@ public boolean storeResponseHeaders(
         BufferPool bufferPool)
     {
         if (responseHeaders.anyMatch(h ->
-            ":status".equals(h.name().asString()) &&
-            h.value().asString().startsWith("2")))
+                (":status".equals(h.name().asString()) &&
+                        Objects.requireNonNull(h.value().asString()).startsWith("2")) ||
+                        Objects.requireNonNull(h.value().asString()).equals("304")))
```

Comment:
The condition would be true if the value is 304 whether header name is ":status" or not. For example "foo: 304" header would be evaluated to true.
You want to OR the last two conditions.